### PR TITLE
Issue 513 configurable timeouts

### PR DESCRIPTION
**Pull request by kartavyashankar**
*Issue #513 :*

*Description of changes:*
Added configurable timeouts for user initiated async api calls

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kartavyashankar** [comment]
Hi @markjschreiber ,
Requesting a review for the changes.
If this looks good, I will procede with changes in the test files.

**Comment by markjschreiber** [comment]
Thanks @kartavyashankar, I will take a look at the code tomorrow.

As there was some minor changes in `main` (bumps of library versions) I rebased this PR onto `main`.

In the meantime, I launched the CI and noticed that there are a few stylecheck failures that you could take a look at https://github.com/awslabs/aws-java-nio-spi-for-s3/actions/runs/10377815249/job/28732827740?pr=517

File: src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java [hunk 1]
```diff
@@ -470,7 +470,7 @@ public void copy(Path source, Path target, CopyOption... options) throws IOExcep
         final var s3Client = s3SourcePath.getFileSystem().client();
         final var sourceBucket = s3SourcePath.bucketName();
 
-        final var timeOut = TIMEOUT_TIME_LENGTH_1;
+        final var timeOut = configuration.getTimeoutLow();
```
Comment (markjschreiber) [inline src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java:473]: Here we should probably change this to use getTimeoutHigh as this was the original issue reported (copy's that take too long time out).

**Review by markjschreiber — CHANGES_REQUESTED** [review]
This looks good. Could you add/ update tests for `S3NioSpiConfiguration` to make sure values are set as expected?

File: src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java (hunk 1 above)
Comment (kartavyashankar) [reply src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java]: Updated the tests and made the required changes. Please take a look when you get time.

Thanks

**Review by markjschreiber — APPROVED** [review]
Looks great thanks!

**Pull request merged**